sycl/unittests/kernel-and-program/PersistentDeviceCodeCache.cpp

```diff
@@ -91,6 +91,8 @@ class PersistenDeviceCodeCache : public ::testing::Test {
       return;
     }
 
+    set_env("SYCL_CACHE_DIR", ".");
+
```

Comment:
I don't think implicitly using current directory to store JIT cash for SYCL unittests is a good approach (especially if the test is launched manually). Is it possible to require explicit cache directory configuration or fail the test?

Reply:
The current directory in this case is <build_root>/tools/sycl/test/test_cache.
I think it is ok to have LIT test artifacts in the build directory.

Reply:
> The current directory in this case is <build_root>/tools/sycl/test/test_cache.

This is wrong assumption in general. I think what you are trying to say is that:
CI system runs the test using `<build_root>/tools/sycl/test/test_cache` as current directory and clears `build_root` directory, so there should be no **additional** side effects from this change.

I agree with that, but that scenario was not my concern. I'm asking about other use cases e.g. developer manually run the test from non-`build_root` directory.

Reply:
I have moved setting the environment variable to LIT infra and add the assertion to guide a user if he runs tests manually.

Reply:
Assertion is NOP in Release. I think we need the test to work in this configuration as well. Right?